Bottom: cefdd35d172882092a04a67cd6e438b0766ac974
Top:    e0e20d9148ab7a7463ac86a01e02423eb14837c0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-09 22:29:34 +0000

Refresh of tmp

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index b9b1e94..9f24340 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -318,7 +318,6 @@ hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double s
   int circular_caps = 0;
   int hairline = 0;
 
-  return;
   CHECK_IS_IN_CONTEXT ();
   if (width == 0.0)
     hairline = 1;
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 46ac706..4ed5821 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1625,11 +1625,9 @@ ghid_draw_everything (BoxTypePtr drawn_area)
 
   save_show_solder = Settings.ShowSolderSide;
 
-  if (reverse_layers)
-    Settings.ShowSolderSide = !Settings.ShowSolderSide;
+  if (!global_view_2d)
+    Settings.ShowSolderSide = reverse_layers;
 
-  if (!global_view_2d && save_show_solder)
-    reverse_layers = !reverse_layers;
   PCB->Data->SILKLAYER.Color = PCB->ElementColor;
   PCB->Data->BACKSILKLAYER.Color = PCB->InvisibleObjectsColor;
 
@@ -1862,7 +1860,7 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   glTranslatef (-widget->allocation.width / 2., -widget->allocation.height / 2., 0);
   glScalef ((ghid_flip_x ? -1. : 1.) / port->zoom,
             (ghid_flip_y ? -1. : 1.) / port->zoom,
-            ((ghid_flip_x == ghid_flip_y) ? 1. :  1.) / port->zoom);
+            ((ghid_flip_x == ghid_flip_y) ? 1. : -1.) / port->zoom);
   glTranslatef (ghid_flip_x ? port->view_x0 - PCB->MaxWidth  :
                              -port->view_x0,
                 ghid_flip_y ? port->view_y0 - PCB->MaxHeight :
